torchvision/prototype/datasets/_api.py

```diff
@@ -23,8 +23,7 @@ def register(dataset: Dataset) -> None:
         register(obj())
 
 
-# This is exposed as 'list', but we avoid that here to not shadow the built-in 'list'
-def _list() -> List[str]:
+def list_names() -> List[str]:
```

Comment:
I went for `list_names` since it corresponds to the `name` parameter of the `load()` function. Alternatives I could think of were more verbose (e.g. `list_available_datasets()`) but I'm happy to hear others thoughts

Reply:
To recap: `list` is a fitting name and this works fine if you do something like

```py
from torchvision.prototype import datasets
datasets.list()
```

but you cannot (or better should not) do

```py
from torchvision.prototype.datasets import list
```

Since we don't want to enforce a specific style on the user, we should use a name that works in both cases.

Reply:
TensorFlow datasets uses [`list_builders`](https://www.tensorflow.org/datasets/api_docs/python/tfds/list_builders) probably for the same reason. That would correspond to `lists_datasets` for us, which I like a little better than `list_names`.